File: src/test/java/ukalus/persistence/Getter.java
/*
 * Created on May 18, 2004
 *  
 */
package ukalus.persistence;

import java.util.function.Function;

/**
 * @author gremopm
 * 
 */
public class Getter implements Function<Reference, Object> {

  private static final long serialVersionUID = 3257283617423241785L;

  public Object apply(Reference object) {
    return object.get();
  }

}

File: src/test/kotlin/ukalus/persistence/Getter.kt
/*
 * Created on May 18, 2004
 *  
 */
package ukalus.persistence

import java.io.Serializable
import java.util.function.Function

/**
 * @author gremopm
 */
class Getter<T> : Function<Reference<T>, Any?>, Serializable {
    override fun apply(`object`: Reference<T>): Any? {
        return `object`.get()
    }
}